src/test_workflow/bwc_test/bwc_test_runner.py

```diff
@@ -0,0 +1,40 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import abc
+import logging
+import os
+
+from system.temporary_directory import TemporaryDirectory
+from test_workflow.test_recorder.test_recorder import TestRecorder
+from test_workflow.test_result.test_suite_results import TestSuiteResults
+
+
+class BwcTestRunner(abc.ABC):
+    def __init__(self, args, test_manifest):
+        self.args = args
+        self.test_manifest = test_manifest
+
+        self.tests_dir = os.path.join(os.getcwd(), "test-results")
+        os.makedirs(self.tests_dir, exist_ok=True)
+        self.test_recorder = TestRecorder(self.args.test_run_id, "bwc-test", self.tests_dir)
+
+    def run(self):
+        with TemporaryDirectory(keep=self.args.keep, chdir=True) as work_dir:
+            all_results = TestSuiteResults()
+            for component in self.components.select(focus=self.args.component):
+                if component.name in self.test_manifest.components:
```

Comment:
Not for this PR, but we do a similar workflow in many places and I don't love the double-select here, first pass `--component name` to pick a component, but then we also only want components that have a test configuration.

One option would be to fail if `self.args.component not in self.test_manifest.components` with an explicit error, and refactor `components.select` to take an array of components, i.e. `self.components.select(focus=self.test_manifest.components)`.

Another, possibly better, could be to do `self.components.select(focus=self.args.component, configs=self.test_manifest.components, key: "bwc_test")` that would return a set of tuples of component + configuration. The code then becomes quite clean:

```python
for component, config in self.components.select(focus=self.args.component, configs: self.test_manifest.components, key: "bwc_test"):
   test_suite = ...
   ...
```

Reply:
Created a cleanup issue here: https://github.com/opensearch-project/opensearch-build/issues/1604 and assigned myself.